src/modules/job-info/job_state.c

```diff
@@ -0,0 +1,479 @@
+/************************************************************\
+ * Copyright 2018 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* job_state.c - store information on state of jobs */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <czmq.h>
+#include <jansson.h>
+#include <flux/core.h>
+
+#include "src/common/libeventlog/eventlog.h"
+
+#include "job_state.h"
+
+/* To handle the common case of user queries on job state, we will
+ * store jobs in three different lists.
+ *
+ * - pending - these are jobs that have not yet in the RUN state, they
+ *   are sorted based on job priority, then job submission time.  -
+ * - running - these are jobs that have transitioned to the RUN state.
+ *   They are sorted by initial run start time.
+ * - inactive - these are jobs that are in the INACTIVE state, they
+ *   are sorted by job completion time.
+ *
+ * There is also an additional list `processing` that stores jobs that
+ * cannot yet be stored on one of the lists above.
+ *
+ * The list `futures` is used to store in process futures.
+ */
+
+struct job_state_ctx {
```

Comment:
Also defined in job_state.h?

Reply:
hmmm strange, maybe I pushed the wrong version of my tree.